pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamConfig.java

```diff
@@ -214,8 +212,9 @@ public static void validateConsumerType(String streamType, Map<String, String> s
     }
   }
 
-  public boolean isServerUploadToDeepStore() {
-    return _serverUploadToDeepStore;
+  public boolean isServerUploadToDeepStore(boolean defaultServerUploadToDeepStore) {
```

Comment:
Directly return the object `Boolean` without passing in the default. We may annotate the return as `@Nullable`

Reply:
Fixed.